app/services/proofing/resolution/plugins/threat_metrix_plugin.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Proofing
+  module Resolution
+    module Plugins
+      class ThreatMetrixPlugin
+        def call(
+          applicant_pii:,
+          current_sp:,
+          request_ip:,
+          threatmetrix_session_id:,
+          timer:,
+          user_email:
+        )
+          unless FeatureManagement.proofing_device_profiling_collecting_enabled?
+            return threatmetrix_disabled_result
+          end
+
+          # The API call will fail without a session ID, so do not attempt to make
+          # it to avoid leaking data when not required.
+          return threatmetrix_id_missing_result if threatmetrix_session_id.blank?
+          return threatmetrix_pii_missing_result if applicant_pii.blank?
+
+          ddp_pii = applicant_pii.dup
+          ddp_pii[:threatmetrix_session_id] = threatmetrix_session_id
+          ddp_pii[:email] = user_email
+          ddp_pii[:request_ip] = request_ip
+
+          timer.time('threatmetrix') do
+            proofer.proof(ddp_pii)
+          end.tap do |result|
+            Db::SpCost::AddSpCost.call(
+              current_sp, :threatmetrix,
+              transaction_id: result.transaction_id
+            )
+          end
+        end
+
+        def proofer
+          @proofer ||=
+            if IdentityConfig.store.lexisnexis_threatmetrix_mock_enabled
+              Proofing::Mock::DdpMockClient.new
+            else
+              Proofing::LexisNexis::Ddp::Proofer.new(
```

Comment:
Just a heads up. Will be updating the config to add a "policy" attribute since we have decided for Authentication we will have a different policy. But otherwise. Taking a look this shouldn't affect our implementation much. may be a merge conflict depending on whats merged first.